src/bytes.rs

```diff
@@ -16,22 +16,50 @@ use crate::{tungstenite::Bytes, Message, WsError};
 /// Every write sends a binary message. If you want to group writes together, consider wrapping
 /// this with a `BufWriter`.
 #[derive(Debug)]
-pub struct ByteWriter<S>(S);
+pub struct ByteWriter<S> {
+    sender: S,
+    state: State,
```

Comment:
The `ByteWriter` changes should probably go into a separate PR? Seem unrelated to the removal of the feature flag